sound/soc/sof/intel/hda-mlink.c

```diff
@@ -96,7 +96,7 @@ struct hdac_ext2_link {
  */
 
 static int hdaml_lnk_enum(struct device *dev, struct hdac_ext2_link *h2link,
-			  void __iomem *ml_addr, int link_idx)
+			  void __iomem *remap_addr, void __iomem *ml_addr, int link_idx)
```

Comment:
Why do you need to pass a separate `ml_addr`? The function uses `ml_addr` or `hlink->ml_addr` interchangeably.

Reply:
the confusion between ml_addr and hlink->ml_addr is something we could add in a separate patch. We still need the ml_addr to figure out where the link descriptors are stored.

Reply:
added a patch